packages/core/src/storage/extract.ts

```diff
@@ -108,3 +149,23 @@ function getTypeMembers(typeDef: StructDefinition | EnumDefinition): TypeItem['m
     return typeDef.members.map(m => m.name);
   }
 }
+
+function getOriginContract(
+  basecontract: ContractDefinition,
```

Comment:
Please rename to `contract`. "Base contract" can be interpreted as "parent contract".